corehq/apps/app_manager/views/view_generic.py

```diff
@@ -59,73 +59,67 @@ def view_generic(request, domain, app_id, module_id=None, form_id=None,
     if form_id and not module_id and module_unique_id is None:
         return bail(request, domain, app_id)
 
-    app = module = form = None
-    try:
-        if app_id:
-            app = get_app(domain, app_id)
+    app = get_app(domain, app_id)
+    module = form = None
 
-        if module_id:
-            try:
-                module = app.get_module(module_id)
-            except ModuleNotFoundException:
-                raise Http404()
-            if not module.unique_id:
-                module.get_or_create_unique_id()
-                app.save()
-        elif module_unique_id:
-            try:
-                module = app.get_module_by_unique_id(module_unique_id)
-            except ModuleNotFoundException:
-                raise Http404()
-            module_id = module.id
-
-        if form_id and module is not None:
-            try:
-                form = module.get_form(form_id)
-            except IndexError:
-                raise Http404()
-        elif form_unique_id:
-            try:
-                form = app.get_form(form_unique_id)
-            except FormNotFoundException:
-                raise Http404()
-            form_id = form.id
+    if module_id:
+        try:
+            module = app.get_module(module_id)
+        except ModuleNotFoundException:
+            raise Http404()
+        if not module.unique_id:
+            module.get_or_create_unique_id()
+            app.save()
+    elif module_unique_id:
+        try:
+            module = app.get_module_by_unique_id(module_unique_id)
+        except ModuleNotFoundException:
+            raise Http404()
+        module_id = module.id
 
-        if form is not None and module is None:
-            # this is the case where only the form_unique_id is given
-            module = form.get_module()
-            module_id = module.id
+    if form_id and module is not None:
+        try:
+            form = module.get_form(form_id)
+        except IndexError:
+            raise Http404()
+    elif form_unique_id:
+        try:
+            form = app.get_form(form_unique_id)
+        except FormNotFoundException:
+            raise Http404()
+        form_id = form.id
 
-    except (ModuleNotFoundException, FormNotFoundException):
-        return bail(request, domain, app_id)
+    if form is not None and module is None:
+        # this is the case where only the form_unique_id is given
```

Comment:
thank you for adding this, whoever did :)

Reply:
@biyeun !